truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/nodes/LoopNode.java

```diff
@@ -167,8 +172,95 @@ protected LoopNode() {
      */
     public static void reportLoopCount(Node source, int iterations) {
         if (CompilerDirectives.inInterpreter()) {
-            ACCESSOR.onLoopCount(source, iterations);
+            if (SUPPORT == null) {
+                final TruffleRuntime runtime = Truffle.getRuntime();
+                SUPPORT = runtime.getCapability(TVMCI.class);
+                if (SUPPORT == null) {
+                    SUPPORT = new NoOp();
+                }
+            }
```

Comment:
What would speak against lazy initialization in the static initializer and make SUPPORT final?
At the moment this SUPPORT field accesses are not thread-safe. We would need double checked locking and SUPPORT to become volatile. If possible we should avoid this.